pennylane/gradients/parameter_shift_cv.py

```diff
@@ -657,8 +665,7 @@ def _update(data):
     # functionality before deprecation.
     diff_methods = tape._grad_method_validation("analytic" if fallback_fn is None else "best")
     all_params_grad_method_zero = all(g == "0" for g in diff_methods)
-
-    if not tape.trainable_params or all_params_grad_method_zero:

```

Comment:
How come `not tape.trainable_params` was removed?

The previous change in this file added a case for `if argnum is None and not tape.trainable_params`.

It seems, however, that we could still have `argnum is not None and not tape.trainable_params`, correct? If so,
* The first `if` that has been added on line 649 would not evaluate to `True`;
* Due to omitting `not tape.trainable_params` on this line (line 661 originally), not evaluating to `True` here either.

Is that considered with the change?

Reply:
> Hmm, in which scenario would argnum be not None but trainable_params be True, argnum = [] ?

Frankly, I'm not sure. Just noticed that that case is not being covered :thinking: 

> It's true I didn't consider such cases, because [parameter_shift](https://github.com/PennyLaneAI/pennylane/blob/e50c386c3abf008623b83366ade0f7b3cb73f9ec/pennylane/gradients/parameter_shift.py#L548-L557) is setup the same way.

Interesting. Would be great to know why `argnum is None` was added there in the first place and why `not tape.trainable_params` did not suffice.

Reply:
It looks like [here](https://github.com/PennyLaneAI/pennylane/commit/7425c5d1df72bd80927762fcf4e48a04ce60db4e) the version I'm using now was introduced to `parameter_shift` 4 months ago, while the version that was present in `parameter_shift_cv` is 6 month old according to the blame. Before that they had the same conditionals, so I'm guessing `parameter_shift_cv` was just overlooked regarding this change.